test/helpers.js

```diff
@@ -160,13 +160,31 @@
       request.on('extractData', function(resp) {
         return resp.data = (resp.httpResponse.body || '').toString();
       });
+      request.on(
+        'error',
+        function (err, response) {
+          response.error = err;
+          let errorCodeMapping = response.request.service.api.errorCodeMapping;
+          if (response.error && response.error.code && errorCodeMapping) {
+            var mapping = errorCodeMapping[response.error.code];
+            if (mapping) {
+              response.error.code = mapping.code;
+              return response.error
+            }
+          }
+          return response.error
+        });
```

Comment:
You don't need to re-implement the listener here. They should be added at https://github.com/aws/aws-sdk-js/blob/c578c8346494654914efb335779ca2620c3e6b36/lib/service.js#L279-L283